### PR TITLE
Implement splitLookup and split.

### PR DESCRIPTION
**Pull request by archblob**
I may have overlooked some opportunities to improve performance.

### CODE REVIEW

File: Data/CritBit/Tree.hs [hunk 1]
```diff
@@ -774,3 +774,65 @@ mapAccumWithKey f start (CritBit root) = second CritBit (go start root)
...
+-- > splitLookup "1" (fromList [("a",5), ("b",3)]) == (empty, fromList [("a",5), ("b",3)])
+-- > splitLookup "b" (fromList [("a",5), ("b",3)]) == (fromList [("a",5)], empty)
+-- > splitLookup "b" (fromList [("a",5), ("c",3)]) == (fromList [("a",5)], fromList [("c",3)])
+-- > splitLookup "a" (fromList [("a",5), ("b",3)]) == (empty, fromList [("b",3])
+-- > splitLookup "c" (fromList [("a",5), ("b",3)]) == (fromList [("a",5), ("b",3)], empty)
+split :: (CritBitKey k, Ord k)
+      => k
+      -> CritBit k v
+      -> (CritBit k v, CritBit k v)
+split = ((\(l, _, r) -> (r,l)) .). splitLookup
```
Comment (bos) [inline Data/CritBit/Tree.hs]: I don't really like this cryptic anonymous function.

Also, please follow the pattern from elsewhere in the source file of explicitly naming the arguments to `split` on the LHS.

File: tests/Properties.hs [hunk 2]
```diff
@@ -310,6 +310,12 @@ t_traverseWithKey _ (KV kvs) = mappedC == mappedM
                   (C.fromList kvs)
         mappedM = Map.toList . runIdentity . Map.traverseWithKey fun $
                   (Map.fromList kvs)
+t_splitLookup :: (CritBitKey k, Ord k) => k -> KV k -> Bool
+t_splitLookup k (KV kvs) = tl C.toList slC == tl Map.toList slM
+  where
+     tl f (l,v,g) = (f l, v,f g)
+     slC = C.splitLookup k $ C.fromList kvs
+     slM  = Map.splitLookup k $ Map.fromList kvs
```
Comment (bos) [inline tests/Properties.hs]: The local variables here add nothing to comprehensibility. I'd rather read this implementation:

``` haskell
t_splitLookup :: (CritBitKey k, Ord k) => k -> KV k -> Bool
t_splitLookup k (KV kvs) =
    fixup C.toList . C.splitLookup k . C.fromList $ kvs == 
    fixup Map.toList . Map.splitLookup k . Map.fromList $ kvs
  where
    fixup f (l,v,g) = (f l, v, f g)
```

**Comment by archblob** [comment]
I have made the changes you requested, i hope it all reads better now.

**Comment by archblob** [comment]
Seeing the implementation of @isturdy I remembered I did not benchmark splitLookup. I rewrote it without using continuations and from 400ns it got down to 17ns , twice as fast as the containers function.

**Comment by isturdy** [comment]
What happened is that your benchmark (for both the CritBit and Map implementation) now returns a tuple that does no real work when evaluated to WHNF--you need to force evaluation of the tuple elements to WHNF as well. I also think that you are missing some optimizations--I am fairly certain that there is no need to rewalk the tree.

**Pull request closed without merging**